File: abstract-factory/java/MySQLDatabaseFactory.java
package AbstractFactory;

// MySQL için ilişkili nesneler üretildi.
public class MySQLDatabaseFactory implements IDatabaseFactory {
    @Override
    public Connection createConnection() {
        return new MySQLConnection();
    }

    @Override
    public Command createCommand() {
        return new MySQLCommand();
    }
}

File: abstract-factory/typescript/mysql-database-factory.ts
import {IDatabaseFactory} from "./database-factory.interface";
import {Command} from "./command";
import {Connection} from "./connection";
import {MysqlCommand} from "./mysql-command";
import {MysqlConnection} from "./mysql-connection";

// MySQL için ilişkili nesneler üretildi.
export class MysqlDatabaseFactory implements IDatabaseFactory{
  createCommand(): Command {
    return new MysqlCommand();
  }

  createConnection(): Connection {
    return new MysqlConnection();
  }
}
